File: src/main/java/nox/core/sys/Win.java
/*
  Copyright (c): 2017 Oleg Sklyar and contributors. License: MIT
 */
package nox.core.sys;

import org.gradle.internal.os.OperatingSystem;


public enum Win {
	gtk("gtk"),
	win32("win32"),
	cocoa("cocoa");

	private final String value;

	Win(String value) {
		this.value = value;
	}

	public static Win current() {
		OperatingSystem os = OperatingSystem.current();
		return os.isWindows() ? win32 : (os.isMacOsX() ? cocoa : gtk);
	}

	@Override
	public String toString() {
		return value;
	}
}

File: src/main/kotlin/nox/sys/Win.kt
/*
  Copyright (c): 2017 Oleg Sklyar and contributors. License: MIT
 */
package nox.sys

import org.gradle.internal.os.OperatingSystem


enum class Win private constructor(private val value: String) {
	gtk("gtk"),
	win32("win32"),
	cocoa("cocoa");

	override fun toString(): String {
		return value
	}

	companion object {

		fun current(): Win {
			val os = OperatingSystem.current()
			return if (os.isWindows) win32 else if (os.isMacOsX) cocoa else gtk
		}
	}
}
